src/item.test.ts

```diff
@@ -0,0 +1,400 @@
+import { ContentType, Tag, TagType, ElementType, AssetType, IBlockElement as BlockElement } from "mapiThriftModels";
+import { fromCapi, Design, Standard, ElementKind, Image } from 'item';
+import { JSDOM } from "jsdom";
+
+const articleContent = {
+    id: "",
+    type: ContentType.ARTICLE,
+    webTitle: "",
+    webUrl: "",
+    apiUrl: "",
+    tags: [],
+    references: [],
+    isHosted: false
+}
+
+const contentWithTag = (tagId: string) => ({
+    ...articleContent,
+    tags: [new Tag({
+        id: tagId,
+        type: TagType.TONE,
+        webTitle: "",
+        webUrl: "",
+        apiUrl: "",
+        references: []
+    })]
+})
+
+const reviewContent = {
+    ...contentWithTag('tone/reviews'),
+    fields: {
+        starRating: 4
+    }
+}
+
+const articleContentWithElement = (element: BlockElement) => ({
+    ...articleContent,
+    blocks: {
+        body: [
+            {
+                id: "",
+                bodyHtml: "",
+                bodyTextSummary: "",
+                attributes: {},
+                published: true,
+                contributors: [],
+                elements: [element]
+            }
+        ]
+    }
+})
+
+const articleContentWithImage = articleContentWithElement({
+    type: ElementType.IMAGE,
+    assets: [
+        {
+            type: AssetType.IMAGE,
+            mimeType: "image/jpeg",
+            file: "https://gu.com/image.jpg",
+            typeData: {
+                aspectRatio: "5:3",
+                width: 5302,
+                height: 3182,
+                isMaster: true
+            }
+        }
+    ],
+    imageTypeData: {
+        caption: "caption",
+        copyright: "",
+        displayCredit: true,
+        credit: "credit",
+        source: "",
+        photographer: "",
+        alt: "alt",
+        mediaId: "",
+        mediaApiUri: "https://image.co.uk",
+        suppliersReference: "",
+        imageType: ""
+    }
+})
+
+const articleContentWithImageWithoutFile = articleContentWithElement({
+    type: ElementType.IMAGE,
+    assets: [
+        {
+            type: AssetType.IMAGE,
+            mimeType: "image/jpeg",
+            file: "",
+            typeData: {
+                aspectRatio: "5:3",
+                width: 5302,
+                height: 3182,
+                isMaster: true
+            }
+        }
+    ],
+    imageTypeData: {
+        copyright: "",
+        source: "",
+        photographer: "",
+        mediaId: "",
+        mediaApiUri: "https://image.co.uk",
+        suppliersReference: "",
+        imageType: ""
+    }
+})
+
+describe('fromCapi returns correct Item', () => {
+    test('media', () => {
+        const item = fromCapi(JSDOM.fragment)(contentWithTag('type/audio'));
```

Comment:
It should be possible to use partial application here to avoid having to call `fromCapi` with all arguments each time:

```ts
const f = fromCapi(JSDOM.fragment);

// Tests

const item = f(contentWithTag('type/audio'));
const item = f(reviewContent);

// etc.
```